### PR TITLE
fix(challenge): Stricter tests for "OOP: Understand Own Properties".

### PR DESCRIPTION
**Pull request by Greenheart**
#### Pre-Submission Checklist

- [x] Your pull request targets the `staging` branch of freeCodeCamp.
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](http://forum.freecodecamp.com/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [x] All new and existing tests pass the command `npm test`. Use `git commit --amend` to amend any fixes.

#### Type of Change

- [x] Small bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)

#### Checklist:

- [x] Tested changes locally.
- [ ] Closes currently open issue (replace XXXX with an issue no): Progress towards #12966 

#### Description

Prevent campers from using the built in `Object.keys()`, since the purpose of the challenge is to use `for...in` combined with `Object.hasOwnProperty()`, to understand what own properties mean, and how to handle them.

**Note:** Breaking change, since it prevents use of `Object.keys()`, which was a valid solution before.

second-test

Ref: https://github.com/freeCodeCamp/freeCodeCamp/issues/12966#issuecomment-277443668

### CODE REVIEW

File: seed/challenges/02-javascript-algorithms-and-data-structures/object-oriented-programming.json [hunk 1]
```diff
@@ -318,7 +318,8 @@
         ""
       ],
       "tests": [
-        "assert(ownProps.includes('name') && ownProps.includes('numLegs'), 'message: <code>ownProps</code> should include the values <code>\"numLegs\"</code> and <code>\"name\"</code>.');"
```
Comment (stayready) [inline seed/challenges/02-javascript-algorithms-and-data-structures/object-oriented-programming.json]: "assert(ownProps.includes('panda') && ownProps.includes(' 2legs'), 'message: <code>ownProps</code>black and whit<code>\"numLegs\"</code> and <code>\"panda"</code>.');"
 
"assert (ownPr

File: seed/challenges/02-javascript-algorithms-and-data-structures/object-oriented-programming.json [hunk 2]
```diff
@@ -318,7 +318,8 @@
...
+        "assert(ownProps.includes('name') && ownProps.includes('numLegs'), 'message: <code>ownProps</code> should include the values <code>\"numLegs\"</code> and <code>\"name\"</code>.');",
```
Comment (stayready) [inline seed/challenges/02-javascript-algorithms-and-data-structures/object-oriented-programming.json]: itional review comment
+        "assert(panda) && ownProps.includes('2Legs'), 'message: <code>ownProps</code> black and white <code>\"2Legs\"</code> and <code>\"panda\"</code>.');",
 
Leave a comment

**Comment by no-stack-dub-sack** [comment]
@Greenheart So what do we need to do since this is a breaking change? Maybe now that the PR with the solutions is merged, you can rebase and update this to include the new solution?? Will that do it?

**Comment by Greenheart** [comment]
@no-stack-dub-sack I don't know if there's much we can do about it. All that will happen is that some campers' solutions that used to work before won't do so anymore. But since this only have been published to beta so far, that actually won't be a problem.

I guess we can see this as a normal fix. Maybe the "breaking changes" checkbox is for breaking features, rather than content?

**Comment by no-stack-dub-sack** [comment]
@Greenheart I think that is more likely the case! As long as this doesn't break the app itself, I think we should be ok, and it doesn't look like it will to me 😄

**Comment by no-stack-dub-sack** [comment]
## LGTM!

tested locally and new changes work

**Pull request merged**